pkg/api/store/workload/workload_store.go

```diff
@@ -410,3 +433,82 @@ func imageUpdated(apiContext *types.APIContext, data map[string]interface{}) (bo
 	}
 	return false, nil
 }
+
+func getSchedulingRule(rule string) (string, string, string) {
```

Comment:
Is it possible to extract and reuse [logic in scheduling mapper](https://github.com/rancher/types/blob/8c6adc8d2fb4c2d7c0f9b2709094b75f24e52005/mapper/scheduling.go#L148)? In case we miss some corner cases using a different parsing implementation